File: redsheet-portal/src/app/database-models/v-long-term-objective-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vLongTermObjectiveRecord {
    constructor(vLongTermObjectiveRecord?: vLongTermObjectiveRecord )  {
        let properties = [ "longTermObjectiveId", "longTermObjectiveValue", "projectBackgroundId",];
        for (let property of properties)
            if (vLongTermObjectiveRecord && vLongTermObjectiveRecord[property])
                this[property] = vLongTermObjectiveRecord[property];
    }
 
	longTermObjectiveId : number =   0 ;
 
    @required()
    @maxLength(1000)
	longTermObjectiveValue : string =   undefined;
 
    @range(1,2147483647)
	projectBackgroundId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vLongTermObjectiveRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vLongTermObjectiveRecords", Schema = "dbo")]
	[Serializable]
	public partial class vLongTermObjectiveRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int LongTermObjectiveId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string LongTermObjectiveValue { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectBackgroundId { get; set; } 

	public vLongTermObjectiveRecord()
	{
    }

    }
}

